pkg/cluster/delete.go

```diff
@@ -251,6 +251,22 @@ func (m *manager) deleteResources(ctx context.Context) error {
 	return nil
 }
 
+func deleteByIdCloudError(err error) error {
+	detailedError, ok := err.(autorest.DetailedError)
+	if ok {
+		switch {
+		case strings.Contains(detailedError.Original.Error(), "CannotDeleteLoadBalancerWithPrivateLinkService"):
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeCannotDeleteLoadBalancerByID,
+				"features.ResourcesClient#DeleteByID", detailedError.Original.Error())
+
+		case strings.Contains(detailedError.Original.Error(), "AuthorizationFailed"):
+			return api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden,
+				"features.ResourcesClient#DeleteByID", detailedError.Original.Error())
+		}
+	}
+	return err
+}
```

Comment:
wondering if this wouldn't have a nicer flow... not asking to change, just wondering. WDYT?
```suggestion
func deleteByIdCloudError(err error) error {
	detailedError, ok := err.(autorest.DetailedError)
	if !ok {
		return err
	}
	switch {
	case strings.Contains(detailedError.Original.Error(), "CannotDeleteLoadBalancerWithPrivateLinkService"):
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeCannotDeleteLoadBalancerByID,
			"features.ResourcesClient#DeleteByID", detailedError.Original.Error())

	case strings.Contains(detailedError.Original.Error(), "AuthorizationFailed"):
		return api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden,
			"features.ResourcesClient#DeleteByID", detailedError.Original.Error())
	default:
		return err	
	}
}
```

Reply:
I think this is nice, but I prefer the one in place as it matches the style of deleteResourcesAndResourceGroup where it checks if ok. The other reason is that there is only one place where we are returning err, and it acts as a default too.

Reply:
having an explicit flow is better than having one less line of code. Also nesting should be avoided when possible. 
https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

Reply:
Fair enough, having a good flow is always nice. We can change to the suggested version then.